### PR TITLE
Support Load Reporter

### PR DESCRIPTION
**Pull request by Polkas**
connected with https://github.com/insightsengineering/teal.reporter/pull/251

My work is done as a collaboration of UCB company with Roche.
insightengineering developers can edit this PR.

Objective: Support the Load Report functionality for `TealRerportCard` and `TealSliceBlock`

- Update the `to_list` and `from_list` `TealSliceBlock` methods. They operate on a string output.
- Export `TealSliceBlock` which is needed for the teal.reporter `ReportCard`.
- `set_id` for the `Reporter` so we protect that it can be reloaded only for the same teal app.
- Fix the wrong docs.

### CODE REVIEW

**Comment by Polkas** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by gogonzo** [comment]
@Polkas could you please merge main to this branch. There are some changes in teal which are important in this context

**Comment by Polkas** [comment]
@gogonzo done:)

File: R/teal_reporter.R [hunk 1]
```diff
@@ -142,21 +146,38 @@ TealSlicesBlock <- R6::R6Class( # nolint: object_name_linter.
...
-    #' @param x (named `list`) with two fields `c("text", "params")`.
-    #' Use the `get_available_params` method to get all possible parameters.
+    #' @description Create the `TealSlicesBlock` from a list.
+    #'
+    #' @param x (`named list`) with two fields `text` and `style`.
+    #' Use the `get_available_styles` method to get all possible styles.
+    #'
     #' @return `self`, invisibly.
+    #' @examples
+    #' TealSlicesBlock <- getFromNamespace("TealSlicesBlock", "teal")
```
Comment (gogonzo) [inline R/teal_reporter.R:156]: don't need to take from Namespace when it is exported

File: R/teal_reporter.R [hunk 2]
```diff
@@ -142,21 +146,38 @@ TealSlicesBlock <- R6::R6Class( # nolint: object_name_linter.
...
+    #' block <- TealSlicesBlock$new()
```
Comment (gogonzo) [inline R/teal_reporter.R:157]: `getFromNamespace` not required as `TealSlicesBlock` is exported

```suggestion
    #' block <- TealSlicesBlock$new()
```

File: R/teal_reporter.R [hunk 3]
```diff
@@ -75,8 +75,12 @@ TealReportCard <- R6::R6Class( # nolint: object_name.
...
 )
 
-#' @title `RcodeBlock`
-#' @keywords internal
+#' @title `TealSlicesBlock`
+#' @docType class
+#' @description
+#' Specialized `TealSlicesBlock` block for managing filter panel content in reports.
+#'
+#' @export
```
[the remark this answers is not in the document]
Comment (Polkas) [reply R/teal_reporter.R:83]: We need to export this class to make it working here https://github.com/Polkas/teal.reporter/blob/7a5c96558f1e25da74e0c67754cb6d793e84041e/R/ReportCard.R#L229

alternatives are:
- transfer TealReporter classes to teal.repoter package
- inbuild in teal.repoter solution like if (class == "TealSlicesBlock") { requireNamspace("teal") TealSlicesBlock <- getFromNamespace("teal", "TealSlicesBlock") }  . I do not want to do that as we build a dependency of teal on teal.reporter, which is lower level.
- ...

[the remark this answers is not in the document]
Comment (gogonzo) [reply R/teal_reporter.R:83]: > * ...

What if `TealReportCard` override `ReportCard$from_list` and handle `TealSlicesBlock`? I'm thinking how `ReportCard` could handle creation of the Blocks as `eval(str2lang(block_class))` executes in `teal.reporter` namespace (and other namespaces including GlobalEnv if not found in teal.reporter). 

The loop through blocks which you pointed out can be overriden somehow. Blocks which are in teal can be "dispatched" in teal namespace and others can be dispatched in `teal.reporter` (via some `super` method, for example `super$dispatch_block`).


This is what I mean. What do you think about this?

In `teal.reporter`

```r
ReporterCard <- R6Class(
   ...
   private = list(
      ...
      dispatch_block = function(block_class) {
         instance <- eval(str2lang(block_class))
      }
   )
)
```

In `teal`
```r
TealReportCard <- R6Class(
     inherits = teal.reporter::ReportCard,
     ...
     private = list(
        dispatch_block = function(block_class) {
            if (block_class %in% <block classes implemented in teal (TealSliceBlock)>) {
              instance <- eval(str2lang(block_class))
            } else {
              super$dispatch_block(block_class)
            }
         }
     )
)
```

[the remark this answers is not in the document]
Comment (Polkas) [reply R/teal_reporter.R]: Great solution:) I applied it on both PRs and seems to work
Most important the TealReportCard has access to unexported TealSlicesBlock - `get("TealSlicesBlock", environment(teal::TealReportCard$private_methods$dispatch_block))`

It is even simpler for both Card classes have the same dispatch_block function. They are the same but different as we gain the proper environment for teal one.

```
      dispatch_block = function(block_class) {
         instance <- eval(str2lang(block_class))
      }
```

[the remark this answers is not in the document]
Comment (gogonzo) [reply R/teal_reporter.R]: amazing

**Review by gogonzo — APPROVED** [review]
Thank you very much for fruitful collaboration 🔥

**Comment by gogonzo** [comment]
@Polkas we need to overcome some cicd issues. You will be able to merge soon 💪

**Comment by Polkas** [comment]
We need to merge the teal.reporter PR first.

File: R/teal_reporter.R [hunk 4]
```diff
@@ -72,11 +72,18 @@ TealReportCard <- R6::R6Class( # nolint: object_name.
...
+  )
 )
 
-#' @title `RcodeBlock`
-#' @keywords internal
+#' @title `TealSlicesBlock`
+#' @docType class
+#' @description
+#' Specialized `TealSlicesBlock` block for managing filter panel content in reports.
+#'
```
Comment (gogonzo) [inline R/teal_reporter.R:86]:
```suggestion
#' @keywords internal
```

**Pull request merged**